meson.build

```diff
@@ -15,32 +16,82 @@ if not meson.is_subproject()
     werror = true
 endif
 
+script = join_paths(meson.source_root(), 'workaround.sh')
+
+r = run_command('/bin/sh', script)
+if r.returncode() != 0
+  message('script failed')
+endif
+
+output = r.stdout().strip()
+errortxt = r.stderr().strip()
+
+prefix = get_option('prefix')
+
+################################################################################
+# Dependencies
+################################################################################
+
+if meson.version().version_compare('>=0.54.0')
+libcjson_dep     = dependency('libcjson',      version: '>=1.7.14', fallback: ['cjson'])
+libcimplog_dep   = dependency('libcimplog')
+libwdmp_dep      = dependency('libwdmp') 
+librtMessage_dep = dependency('librtMessage')
+librbuscore_dep  = dependency('librbuscore')
+librbus_dep      = dependency('librbus')
+libjansson_dep = dependency('libjansson')
+else
+warning('You are running an older version of meson.')
+warning('In order for dependency management to be supported you need a newer version of meson (0.54+)')
+# For older versions of meson, you need to have the upstream dependencies installed
+# in the correct locations.
+libcjson_dep     = dependency('libcjson')
+libcimplog_dep   = dependency('libcimplog')
+libwdmp_dep      = dependency('libwdmp') 
+librtMessage_dep = dependency('librtMessage')
+librbuscore_dep  = dependency('librbuscore')
+librbus_dep      = dependency('librbus')
+libjansson_dep = dependency('libjansson')
+endif
+
+all_dep = [libcjson_dep, librtMessage_dep, librbuscore_dep, librbus_dep, libcimplog_dep, libwdmp_dep, libjansson_dep]
+
 ################################################################################
 # Common variable definitions
 ################################################################################
 
-inc_base = 'include/'+meson.project_name()
+inc_base = ['include/'+meson.project_name()]
 
 ################################################################################
 # Generate the version header file
 ################################################################################
 
 subdir('include/'+meson.project_name())
 
-install_headers([headers, ver_h], subdir: meson.project_name())
+install_headers(['include/'+meson.project_name()+'/cpeabs.h', ver_h], subdir: meson.project_name())
 
 ################################################################################
 # Define the libraries
 ################################################################################
 
-inc = include_directories(inc_base)
+inc = include_directories([inc_base])
 
+src_args = ['-lcjson', '-lpthread']
```

Comment:
The former is already handled by `libcjson_dep`, the latter should not be handled by manual -l flags, see https://mesonbuild.com/FAQ.html#what-is-the-correct-way-to-use-threads-such-as-pthreads